omnibus/cookbooks/omnibus-supermarket/spec/libraries/config_spec.rb

```diff
@@ -75,10 +92,146 @@
             's3_region' => s3_region,
             's3_domain_style' => s3_domain_style
             })
-          expect{ described_class.audit_config(incompatible_config) }
+          expect{ described_class.audit_s3_config(incompatible_config) }
             .to raise_error(Supermarket::Config::IncompatibleConfig)
         end
       end
     end
   end
+
+  describe '#audit_fips_config' do
+    let(:default_config) do
+      { 'fips_enabled' => nil }
+    end
+
+    context 'when the installer was built to support FIPS' do
+      it 'does not raise' do
+        expect(described_class).to receive(:built_with_fips?).and_return(true)
+
+        expect{ described_class.audit_fips_config(default_config) }
```

Comment:
Nit: I prefer to avoid explicit "this didn't happen" expectations - could we instead invoke `audit_fips_config` and rely on a raised exception to cause the test to fail?

Reply:
Right, I mean having the test consist of:
`described_class.audit_fips_config` without the surrounding `expect`. If it does erroneously raise, the test will fail without using `not_raise`

Reply:
Oooooh. 🤔 Do you think the "it does not raise" statement is sufficient for establishing the expectation for the reader? I tend to skim a spec looking for `expect` statements to find out what's being tested.

Reply:
I'll simplify this down to just the method call.